src/fsharp/FSharp.Build/Microsoft.FSharp.NetSdk.targets

```diff
@@ -45,54 +45,24 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
 
   <PropertyGroup Condition="'$(PackProjectInputFile)' == ''">
     <PackProjectInputFile>$(MSBuildProjectFullPath)</PackProjectInputFile>
-  </PropertyGroup>    
-
-  <!--
-    Always reference FSharp.core for coreclr builds
-    TargetFSharpCorePackageIdentity = Name of package to reference      (Default = Microsoft.FSharp.Core.netcore)
-    TargetFSharpCorePackageVersion  = Version of package to reference   (Default = TargetFramework = netcoreapp1.0-1.1 netstandard1.0-1.6 then '1.0.0-*')
-                                                                        (Default = TargetFramework = netcoreapp2.0 netstandard2.0         then '1.0.0-*')
-  -->
-  <PropertyGroup Condition=" '$(TargetFSharpCorePackageVersion)' == '' ">
-    <TargetFSharpCorePackageVersion Condition=" $(TargetFramework.StartsWith(netcoreapp1.)) or $(TargetFramework.StartsWith(netstandard1.)) ">1.0.0-*</TargetFSharpCorePackageVersion>
-    <TargetFSharpCorePackageVersion Condition=" $(TargetFramework.StartsWith(netcoreapp2.)) or $(TargetFramework.StartsWith(netstandard2.)) ">1.0.0-*</TargetFSharpCorePackageVersion>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(TargetFSharpCorePackageIdentity)' == '' ">
-    <TargetFSharpCorePackageIdentity Condition=" '$(TargetFrameworkIdentifier)' != '.NETFramework' " >Microsoft.FSharp.Core.netcore</TargetFSharpCorePackageIdentity>
+  <PropertyGroup Condition=" '$(TargetFSharpCorePackageVersion)' == '' ">
+    <TargetFSharpCorePackageVersion>4.2.0-*</TargetFSharpCorePackageVersion>
```

Comment:
As an important note about design (got bitten by my wrong assumptions before):

- `4.2.0-*` will resolve `4.2.0` when is released, so is not restricted to prerelease
- `4.2.0-*` will **not** resolve `4.2.1`

allow patch version is choice (because make easier to release a fix of `fsharp.core` without change templates or sdk), just want to be sure 

An example to try it: 
- create a console app
- add `<PackageReference Include="Paket" Version="4.0.0-*" />` (used because is a sample nupkg with prerelease deps)
- `Paket` nupkg as has versions:
  - `4.0.1`
  - `4.0.0`
  - `4.0.0-rc9002`

So `4.0.0-*` will resolve `4.0.0` but not `4.0.1`